tests/cases/issues/issue_4163_flickable_parent_percent.slint

```diff
@@ -0,0 +1,60 @@
+import { ListView } from "std-widgets.slint";
+// Copyright © SixtyFPS GmbH <info@slint.dev>
+// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-1.1 OR LicenseRef-Slint-commercial
+
+component ShowResult inherits Rectangle {
+    width: 50%;
+    height: 100%;
+
+    VerticalLayout {
+        ListView {
+            for file[idx] in [1,2,3]:Rectangle {
+                height: 20px;
+                // width: parent.width;    // change to this line compiles fine
+                width: 100%;  // An error occurred!!!
+            }
+        }
+    }
+}
+export component TestCase inherits Window {
+    width: 326px;
+    height: 326px;
+
+    HorizontalLayout {
+        width: 100%;
+        height: 100%;
+
+        ShowResult {}
+    }
+
+
+    fli := Flickable {
+        viewport-height: 1000px;
+        viewport-width: 2000px;
+        rec := Rectangle {
+            width: 100%;
+            height: parent.height;
+        }
+    }
+
+    out property <bool> test:
+        // The % applies to the viewport
```

Comment:
Note that our terminology is wrong: the viewport is the "portion" that's visible. So the % applies to the content width, not the viewport width - while parent refers to the content.

The inconsistency is not good.

Since the % asserts right now, I guess aligning it with parent would be the "compatible" fix.

I'm inclined to say that parent referring to the content size would make more sense to me, as the Flickable can always be referred to by name. But I think that would be a breaking change.

Reply:
> Note that our terminology is wrong: the viewport is the "portion" that's visible. So the % applies to the content width, not the viewport width - while parent refers to the content.
>
> The inconsistency is not good.

parent doesn't refer to the content, it refer to the flickable itself.

"content" and "viewport" are two name for the same things. We seem to use viewport consistently.

But it's true that you already filled an issue to replace that. https://github.com/slint-ui/slint/issues/1443  
But that's out of scope of this PR